src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java

```diff
@@ -39,6 +39,28 @@ public BadArgumentValue(String message) {
         }
     }
 
+    public static class OutOfRangeArgumentValue extends BadArgumentValue {
+        private static final long serialVersionUID = 0L;
+
+        public OutOfRangeArgumentValue(final String arg, final double minValue, final double maxValue, final double value) {
+            super(arg, Double.toString(value), getMessage(minValue, maxValue));
+        }
+
+        private static String getMessage(final double minValue, final double maxValue) {
+            final boolean outMinValue = minValue != Double.NEGATIVE_INFINITY;
+            final boolean outMaxValue = maxValue != Double.POSITIVE_INFINITY;
+            if (outMinValue && outMaxValue) {
+                return String.format("allowed range (%.2f, %.2f).", minValue, maxValue);
+            } else if (outMinValue) {
+                return String.format("minimum allowed value %.2f", minValue);
+            } else if (outMaxValue) {
+                return String.format("maximum allowed value %.2f", maxValue);
+            }
+            return "unkwnon";
```

Comment:
Changed, although this point should never be reached. Maybe is better to thrown an `IllegalArgumentException` if so?